components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/service/WebSubHubAdapterServiceImpl.java

```diff
@@ -0,0 +1,329 @@
+            throws WebSubAdapterServerException {
+
+        try {
+            URIBuilder uriBuilder = new URIBuilder(webSubHubBaseUrl);
+            uriBuilder.addParameter(HUB_MODE, operation);
+            uriBuilder.addParameter(HUB_TOPIC, topic);
+            return uriBuilder.build().toString();
+        } catch (URISyntaxException e) {
+            log.error("Error building URL", e);
+            throw handleServerException(ERROR_INVALID_WEB_SUB_HUB_BASE_URL, e);
+        }
+    }
+
+    private void makeTopicMgtAPICall(String topic, String webSubHubBaseUrl, String operation)
+            throws WebSubAdapterException {
+
+        String topicMgtUrl = buildURL(topic, webSubHubBaseUrl, operation);
+
+        ClientManager clientManager = WebSubHubAdapterDataHolder.getInstance().getClientManager();
+        HttpPost httpPost = clientManager.createHttpPost(topicMgtUrl, null);
+
+        WebSubHubCorrelationLogUtils.triggerCorrelationLogForRequest(httpPost);
+        final long requestStartTime = System.currentTimeMillis();
+
+        CompletableFuture<HttpResponse> future = clientManager.executeAsync(httpPost);
+
+        future.thenAccept(response -> {
+            try {
+                handleTopicMgtResponse((CloseableHttpResponse) response, httpPost, topic, operation, requestStartTime);
+            } catch (IOException | WebSubAdapterException e) {
+                log.error("Handling topic management response failed. ", e);
+            }
+        }).exceptionally(ex -> {
+            log.error("Topic management API call failed. ", ex);
+            return null;
+        });
+    }
+
+    private void handleTopicMgtResponse(CloseableHttpResponse response, HttpPost httpPost,
+                                               String topic, String operation, long requestStartTime)
+            throws IOException, WebSubAdapterException {
+
+        StatusLine statusLine = response.getStatusLine();
+        int responseCode = statusLine.getStatusCode();
+        String responsePhrase = statusLine.getReasonPhrase();
+
+        if (responseCode == HttpStatus.SC_OK) {
+            handleSuccessfulTopicMgt(response, httpPost, topic, operation, requestStartTime,
+                    responseCode, responsePhrase);
+        } else if ((responseCode == HttpStatus.SC_CONFLICT && operation.equals(REGISTER)) ||
+                (responseCode == HttpStatus.SC_NOT_FOUND && operation.equals(DEREGISTER))) {
+            handleConflictOrNotFound(response, httpPost, topic, operation, requestStartTime,
+                    responseCode, responsePhrase);
+        } else {
+            handleFailedTopicMgt(response, httpPost, topic, operation, requestStartTime,
+                    responseCode, responsePhrase);
+        }
+    }
+
+    private static void handleAsyncResponse(HttpResponse response, HttpPost request, long requestStartTime,
+                                             EventContext eventContext, String url, String topic) {
```

Comment:
Not formatted

Reply:
Addressed with https://github.com/wso2-extensions/identity-event-publishers/pull/1/commits/7999427bc5211016a93107fb27f9efd67513d525